### PR TITLE
Unit tests for Calling extension methods on obsolete types and modules

### LINKED ISSUE #13269: Calling extension methods on obsolete types/modules should generate appropriate warning.
Make sure, calling extension methods on the types/modules which have ObsoleteAttribute on them  generate appropriate warning.

**Update**: probably also makes sense to verify what C# does.

### PR DESCRIPTION
**Pull request by edgarfgp**
This PR adds unit test to verify the existing behavior when calling Calling extension methods on obsolete types and modules . Fixes #13269 .

This test uses the **Obsolete("Use B instead", true)** configuration that will produce compiler errors

We could also test **Obsolete**  that will produce warning.  NOT done in this PR (But happy to do it ) :)

### CODE REVIEW

File: tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs [hunk 1]
```diff
@@ -263,4 +262,281 @@ let c = Color.Red
...
+type ButtonExtensions =
+
+    [<Extension>]
+    static member inline text(this: Button, text: string) =
+        { this with Text = text }
+        """
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withErrorCodes [ 101; 101]
```
Comment (KevinRansom) [inline tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs:333]: Please could you use withDiagnosticErrors like here:
https://github.com/dotnet/fsharp/blob/a65ace7698c159c34bd00f6408c7d4beb89b687d/tests/FSharp.Compiler.ComponentTests/Conformance/Expressions/BindingExpressions/BindingExpressions.fs#L24

````
        |> withDiagnostics [
            (Error 222, Line 6, Col 1, Line 7, Col 1, "Files in libraries or multiple-file applications must begin with a namespace or module declaration, e.g. 'namespace SomeNamespace.SubNamespace' or 'module SomeNamespace.SomeModule'. Only the last source file of an application may omit such a declaration.")
        ]
````

Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs]: Sure . I updated the rest of the test for consistency :)

**Comment by edgarfgp** [comment]
This is ready for a second review :)

**Review by vzarytovskii — APPROVED** [review]
That's great! Thanks for that!

**Review by psfinaki — APPROVED** [review]
A bunch of typos you can safely ignore :) Thanks a lot, it was very educational to look at all the possible edge cases.

File: tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs [hunk 2]
```diff
@@ -263,4 +265,272 @@ let c = Color.Red
...
+        """
+        |> ignoreWarnings
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 101, Line 10, Col 20, Line 10, Col 36, "This construct is deprecated. Use B instead")
+        ]
+
+    [<Fact>]
+    let ``Obsolete attribute is taken into account when used on an type property and set property via module using an extesnion method`` () =
```
[the remark this answers is not in the document]
Comment (edgarfgp) [reply tests/FSharp.Compiler.ComponentTests/Language/AttributeCheckingTests.fs]: Thanks for the review :)

**Review by KevinRansom — APPROVED** [review]
Thanks for taking care of this


Kevin
